### PR TITLE
MRG: Add multiple pos and sensors

### PR DESCRIPTION
**Pull request by larsoner**
This updates our `plot_head_positions` function to make it more usable for diagnostic plotting.

### CODE REVIEW

**Comment by larsoner** [comment]
Updated plots:

https://6614-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.zrmDOPJ/html/auto_examples/preprocessing/plot_movement_compensation.html

**Comment by larsoner** [comment]
Cleaned up:

https://6615-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.ML5RvSq/html/auto_examples/preprocessing/plot_movement_compensation.html

File: examples/preprocessing/plot_movement_compensation.py [hunk 1]
```diff
@@ -30,12 +30,22 @@
...
-mne.viz.plot_head_positions(pos, mode='traces')
+mne.viz.plot_head_positions(
+    pos, mode='traces', destination=raw.info['dev_head_t'], info=raw.info)
 
 ###############################################################################
-# Process our simulated raw data (taking into account head movements)
+# This can also be visualized using a quiver.
+
+mne.viz.plot_head_positions(
+    pos, mode='field', destination=raw. info['dev_head_t'], info=raw.info)
```
Comment (agramfort) [inline examples/preprocessing/plot_movement_compensation.py]: extra space before info

File: examples/preprocessing/plot_movement_compensation.py [hunk 2]
```diff
@@ -44,15 +54,18 @@
...
 
-# 0. Take average of stationary data (bilateral auditory patterns)
+###############################################################################
+# First, take the average of stationary data (bilateral auditory patterns).
 evoked_stat = mne.Epochs(raw_stat, events, 1, -0.2, 0.8).average()
 evoked_stat.plot_topomap(title='Stationary', **topo_kwargs)
 
-# 1. Take a naive average (smears activity)
+###############################################################################
+# Second, tak a naive average (smears activity).
```
Comment (agramfort) [inline examples/preprocessing/plot_movement_compensation.py]: take

File: examples/preprocessing/plot_movement_compensation.py (hunk 1 above)
Comment (agramfort) [reply examples/preprocessing/plot_movement_compensation.py]: can you call pos head_pos to be explicit?

File: mne/chpi.py [hunk 3]
```diff
@@ -478,10 +478,13 @@ def _fit_cHPI_amplitudes(raw, time_sl, hpi, fit_time, verbose=None):
 
     data_diff = np.dot(model, X).T - this_data
 
-    # compute amplitude correlation (For logging)
-    g_sin = 1 - np.sqrt((data_diff**2).sum() / (this_data**2).sum())
-    g_chan = 1 - np.sqrt((data_diff**2).sum(axis=1) /
-                         (this_data**2).sum(axis=1))
+    # compute amplitude correlation (for logging), protect against zero
+    norm = (this_data ** 2).sum(axis=1)
```
Comment (agramfort) [inline mne/chpi.py]: if this_data is not too small I suspect

norm = np.linalg.norm(this_data, axis=1)**2

avoid the allocation of this_data ** 2

File: examples/preprocessing/plot_movement_compensation.py (hunk 2 above)
Comment (mmagnuski) [reply examples/preprocessing/plot_movement_compensation.py]: apart from 'take' I'd be a little more verbose in this point (mentioning that the average is taken from the dataset with simulated movement, without movement compensation which results in smeared activity).

**Comment by agramfort** [comment]
why is there no amplitude in colobar at https://6615-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.ML5RvSq/html/_images/sphx_glr_plot_movement_compensation_006.png ?

**Comment by mmagnuski** [comment]
@agramfort If you are asking about the ticks that may be how they are now placed in agreement with topo contours. And the first topo has only contours at zero, should be an easy fix if that is the problem.

**Comment by larsoner** [comment]
@mmagnuski that sounds like a likely culprit. If we don't already, it seems like we should use the same level contours for all subplots within a given figure if possible. Do you have time to look (in a separate PR)?

Other comments addressed by latest commit.

**Comment by larsoner** [comment]
@mmagnuski do you approve, too?

**Comment by mmagnuski** [comment]
@larsoner Yes I can take a look this week.
I do approve, but as I never used MEG I might not be the best audience to judge. The example looks beautiful anyway :)

**Pull request merged**